test/performance/scale_revision_by_load_test.go

```diff
@@ -0,0 +1,215 @@
+			scaleRevisionByLoad(t, numClients, tc)
+		})
+	}
+	if err := testgrid.CreateXMLOutput(tc, t.Name()); err != nil {
+		t.Fatalf("Cannot create output XML: %v", err)
+	}
+}
+
+func scaleRevisionByLoad(t *testing.T, numClients int, tc []junit.TestCase) {
+	perfClients, err := Setup(t)
+	if err != nil {
+		t.Fatalf("Cannot initialize performance client: %v", err)
+	}
+
+	names := test.ResourceNames{
+		Service: test.ObjectNameForTest(t),
+		Image:   "observed-concurrency",
+	}
+	clients := perfClients.E2EClients
+
+	defer TearDown(perfClients, names, t.Logf)
+	test.CleanupOnInterrupt(func() { TearDown(perfClients, names, t.Logf) })
+
+	t.Log("Creating a new Service")
+	objs, err := test.CreateRunLatestServiceReady(t, clients, &names, &test.Options{
+		ContainerResources: corev1.ResourceRequirements{
+			Requests: corev1.ResourceList{
+				corev1.ResourceCPU:    resource.MustParse("50m"),
+				corev1.ResourceMemory: resource.MustParse("20Mi"),
+			},
+		}},
+		func(service *v1alpha1.Service) {
+			service.Spec.RunLatest.Configuration.RevisionTemplate.Annotations = map[string]string{"autoscaling.knative.dev/target": fmt.Sprintf("%d", containerConcurrency)}
+		},
+	)
+	if err != nil {
+		t.Fatalf("Failed to create Service: %v", err)
+	}
+
+	domain := objs.Route.Status.Domain
+	endpoint, err := ingress.GetIngressEndpoint(clients.KubeClient.Kube)
+	if err != nil {
+		t.Fatalf("Cannot get service endpoint: %v", err)
+	}
+
+	// Make sure we are ready to serve.
+	st := time.Now()
+	t.Log("Starting to probe the endpoint at", st)
+	_, err = pkgTest.WaitForEndpointState(
+		clients.KubeClient,
+		t.Logf,
+		domain+"/?timeout=10", // To generate any kind of a valid response.
+		test.RetryingRouteInconsistency(func(resp *spoof.Response) (bool, error) {
+			_, _, err := parseResponse(string(resp.Body))
+			return err == nil, nil
+		}),
+		"WaitForEndpointToServeText",
+		test.ServingFlags.ResolvableDomain)
+	if err != nil {
+		t.Fatalf("The endpoint at domain %s didn't serve the expected response: %v", domain, err)
+	}
+	t.Logf("Took %v for the endpoint to start serving", time.Since(st))
+
+	scaleChannel := make(chan *scaleEvent)
+	endMonitoring := make(chan struct{})
+	eg := errgroup.Group{}
+
+	eg.Go(func() error {
+		t.Logf("Scale Monitoring Started")
+		defer close(scaleChannel)
+		previousNumEndpoints := 1 //We start with 1 pod running
+		for {
+			select {
+			case <-endMonitoring:
+				t.Log("Scale Monitoring Finished")
+				return nil
+			default:
+				currentNumEndpoints, err := test.NumEndpoints(clients, test.ServingNamespace, names.Service)
+				if err != nil {
+					t.Logf("Unable to get pod count: %v", err.Error())
+				}
+				if currentNumEndpoints != previousNumEndpoints {
+					event := &scaleEvent{
+						oldScale:  previousNumEndpoints,
+						newScale:  currentNumEndpoints,
+						timestamp: time.Now(),
+					}
+					scaleChannel <- event
+					previousNumEndpoints = currentNumEndpoints
+				}
+				time.Sleep(100 * time.Millisecond)
```

Comment:
The granularity of the scale event will be determined by this. Is it possible at all to run the test based on an informer?